pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -1039,8 +1040,7 @@ void ConsumerImpl::hasMessageAvailableAsync(HasMessageAvailableCallback callback
         return;
     }
 
-    BrokerGetLastMessageIdCallback callback1 = [this, lastDequed, callback](Result result,
-                                                                            MessageId messageId) {
+    getLastMessageIdAsync([this, lastDequed, callback](Result result, MessageId messageId) {
         if (result == ResultOk) {
```

Comment:
nit: can you refactor it to 
```
const auto ret = (result == ResultOk && messageId > lastDequed && messageId.entryId() != -1);
callback(result, ret);
```
or even not use ret.

Reply:
I don't think that this would improve readability